docs/backend/widgets.md

```diff
@@ -12,15 +12,652 @@ myst:
+                           onkeydown view/onkeydown;
+                           onkeyup view/onkeyup;
+                           value view/value;
+                           disabled view/disabled;
+                           tabindex view/tabindex;
+                           onfocus view/onfocus;
+                           onblur view/onblur;
+                           onchange view/onchange;
+                           readonly view/readonly;
+                           alt view/alt;
+                           accesskey view/accesskey;
+                           onselect view/onselect;
+                           size view/size;
+                           maxlength view/maxlength;
+                           placeholder view/placeholder;
+                           autocapitalize view/autocapitalize;" />
+</html>
+```
+
+Now you can override the template factory in the `updateWidgets()` method of your form class.
+
+```python
+from zope.browserpage.viewpagetemplatefile import ViewPageTemplateFile as Z3ViewPageTemplateFile
+from z3c.form.interfaces import INPUT_MODE
+
+
+class AddForm(DefaultAddForm):
+
+    def updateWidgets(self, prefix=None):
+        """ """
+        # Call parent to set-up initial widget data
+        super().updateWidgets(self, prefix=prefix)
+
+        # Note we need to be discreet to different form modes (view, edit, hidden)
+        if self.fields["sections"].mode == INPUT_MODE:
+
+            # Modify a widget with certain name for our purposes
+            widget = self.widgets["sections"]
+
+            # widget.template is a template factory -
+            # Widget.render() will associate later this factory with the widget
+            widget.template = Z3ViewPageTemplateFile("templates/sections.pt")
+```
+
+You can also interact with your `form` class instance from the widget template:
+
+```html
+<!-- Some hidden JSON data for our JavaScript by calling a method on our form class -->
+<span style="display:none" tal:content="view/form/getBlockPlanJSON" />
+```
+
+
+### Set a template for your own widget type
+
+You can set the template used by the widget with the `<z3c:widgetTemplate>` ZCML directive.
+
+```xml
+    <z3c:widgetTemplate
+        mode="display"
+        widget=".interfaces.INamedFileWidget"
+        layer="z3c.form.interfaces.IFormLayer"
+        template="file_display.pt"
+        />
+```
+
+
+### Widget frame override
+
+You can override widget templates as instructed for `z3c.form`.
+`plone.app.z3cform` renders [a frame around each widget](https://github.com/plone/plone.app.z3cform/blob/master/plone/app/z3cform/templates/widget.pt), which usually consists of:
+
+-   Label
+-   Required marker
+-   Description
+
+You might want to customize this widget frame for your own form.
+Below is an example of how to do it.
+
+Copy [`widget.pt`](https://github.com/plone/plone.app.z3cform/blob/master/plone/app/z3cform/templates/widget.pt) to your own package, and edit it.
```

Comment:
The subsequent code example names a template as `demo-widget.pt`. Should this line mention that name?

Also what is the file path to the template in the package?